pkg/meta/base.go

```diff
@@ -1999,17 +1999,30 @@ func (m *baseMeta) Check(ctx Context, fpath string, repair bool, recursive bool,
 					statBroken = true
 				}
 
+				if statAll {
+					s, st := m.calcDirStat(ctx, inode)
```

Comment:
this slows the repair since we call the `calcDirStat` again within the `doSyncDirStat` func in L2017.

Reply:
fixed